flocker/node/functional/test_deploy.py

```diff
@@ -32,12 +32,13 @@ class P2PNodeDeployer(object):
     https://clusterhq.atlassian.net/browse/FLOC-1732
     """
     def __init__(self, hostname, volume_service, docker_client=None,
-                 network=None):
+                 network=None, node_uuid=None):
```

Comment:
How many tests are left that *don't* supply a `node_uuid`? It seems like you've updated most of them in this branch.
If so, why not just make this a required parameter and remove the warning from `ApplicationNodeDeployer.__init__` and other IDeployer initialisers?